core/src/test/java/org/apache/calcite/test/TypeCoercionTest.java

```diff
@@ -105,21 +105,21 @@ private static ImmutableList<RelDataType> combine(
     final Fixture f = fixture();
     f.checkCommonType(f.nullType, f.nullType, f.nullType, true);
     // BOOLEAN
-    f.checkCommonType(f.nullType, f.booleanType, f.booleanType, true);
+    f.checkCommonType(f.nullType, f.booleanType, f.nullableBooleanType, true);
```

Comment:
@caicancai Please check this test. Before this PR, we thought NULL and Boolean's common type was Boolean. After this PR. The common types will be nullableBoolean.